tests/widgets_block.rs

```diff
@@ -46,6 +49,80 @@ fn widgets_block_renders() {
     terminal.backend().assert_buffer(&expected);
 }
 
+#[test]
+fn widgets_block_titles_overlap() {
+    let test_case = |block, area: Rect, expected| {
+        let backend = TestBackend::new(area.width, area.height);
+        let mut terminal = Terminal::new(backend).unwrap();
+        terminal
+            .draw(|f| {
+                f.render_widget(block, area);
+            })
+            .unwrap();
+        terminal.backend().assert_buffer(&expected);
+    };
+
+    // Left overides the center
+    test_case(
+        Block::default()
+            .title(Title::from("aaaaa").alignment(Alignment::Left))
+            .title(Title::from("bbb").alignment(Alignment::Center))
+            .title(Title::from("ccc").alignment(Alignment::Right)),
+        Rect {
+            x: 0,
+            y: 0,
+            width: 10,
+            height: 1,
+        },
+        Buffer::with_lines(vec!["aaaaab ccc"]),
+    );
+
+    // Left alignment overides the center alignment which overides the right alignment
+    test_case(
+        Block::default()
+            .title(Title::from("aaaaa").alignment(Alignment::Left))
+            .title(Title::from("bbbbb").alignment(Alignment::Center))
+            .title(Title::from("ccccc").alignment(Alignment::Right)),
+        Rect {
+            x: 0,
+            y: 0,
+            width: 11,
+            height: 1,
+        },
+        Buffer::with_lines(vec!["aaaaabbbccc"]),
+    );
+
+    // Multiple left alignment overides the center alignment and the right alignment
+    test_case(
+        Block::default()
+            .title(Title::from("aaaaa").alignment(Alignment::Left))
+            .title(Title::from("aaaaa").alignment(Alignment::Left))
+            .title(Title::from("bbbbb").alignment(Alignment::Center))
+            .title(Title::from("ccccc").alignment(Alignment::Right)),
+        Rect {
+            x: 0,
+            y: 0,
+            width: 11,
+            height: 1,
+        },
+        Buffer::with_lines(vec!["aaaaabaaaaa"]),
```

Comment:
This is a really odd result!
It happens because we're not clearing the space between each title.
I'm not sure it's worth fixing. We could probably leave this and worry about it if someone actually ever hits it.

Reply:
What do you believe should happen instead?

Reply:
I'm not 100% certain, but probably here "aaaaa aaaaa" would be kinda right?

Reply:
Well the behavior I set was the following: left titles override the center titles which override the right alignments. As such, when the space is too small, people might be able to grasp where other titles are if they aren't overwritten. I didn't want to declare spaces between titles as it might remove some information for the user.

Reply:
Yeah, I think this is the sort of thing that we can probably defer until someone gets impacted by it and work it out based on a real world application.

Reply:
yep - it happens to the best of us :D